Gdal 3.9.1

dotysan opened the pull request:
See commit log.

Most notably, review this patch and make sure we haven't disabled any drivers you want to use. https://github.com/dotysan/python-gis/commit/65d015a61daf0a7c52bb856ef39f0c511721c1dd#diff-afd06829678bc2afbdd90ad5343b244b9a42735ca01b1dcd3395a0f312e1c50c

@mpalmerlee @r2voo @TonyJRamos @RMendosa @Kbriggs94

dotysan committed:
Bump ffmpeg-static from Feb 26 to Jun 4 revision

dotysan committed:
Bump GDAL to 3.9.1

dotysan committed:
Replace Poppler with PDFium for massive performance boost

 - also speed up build by removing tests

dotysan committed:
Bump NumPy to v2.0

 - also save ldd sanity checks
 - also save all format details
 - oops, forgot to check in the *.txt logs

dotysan committed:
Try to intelligently login to container registries

dotysan committed:
Remove dependencies that drag in system PROJ

dotysan committed:
Do not build with test deps

dotysan committed:
Replace system PROJ with newer build from source

 - update txt files

dotysan committed:
Expressly enable some GDAL drivers

 - also move COPY after apt commands
 - also fix line formatting bug in all-formats.txt

dotysan committed:
Lean ffmpeg-build with debian-slim

 - also converge all the apt into one RUN

dotysan committed:
Lean build-gdal with debian-slim
    - also converge all apt-get into one RUN
    - explicity add some deps back in

dotysan committed:
Build JPEG XL 0.10 from source rather than system 0.7

 - explicity enable a bunch more drivers
 - using system LittleCMS for now

dotysan committed:
Minor updates from rebuild

dotysan committed:
More hard-coded GDAL drivers enabled

 - disable HDF5 due to threadding
 - use system brotli

dotysan committed:
Disable all implicit drivers

dotysan committed:
Add GNU Awk

dotysan committed:
Add some commented suggestions back in

mpalmerlee commented on ffinfo.txt:
```diff
@@ -1,6 +1,6 @@
 ffmpeg version 6.1.1 Copyright (c) 2000-2023 the FFmpeg developers
 built with gcc 12 (Debian 12.2.0-14)
-configuration: --ld=g++ --prefix=/ffmpeg-static/build/20240402 --pkg-config-flags=--static --extra-cflags=-I/ffmpeg-static/build/20240402/include --extra-ldflags=-L/ffmpeg-static/build/20240402/lib --extra-libs='-lpthread -lm' --bindir=/ffmpeg-static/build/20240402/bin --disable-autodetect --disable-ffprobe --disable-everything --disable-doc --enable-gpl --enable-libx264 --enable-decoder='h264,mp ...
+configuration: --ld=g++ --prefix=/ffmpeg-static/build/20240813 --pkg-config-flags=--static --extra-cflags=-I/ffmpeg-static/build/20240813/include --extra-ldflags=-L/ffmpeg-static/build/20240813/lib --extra-libs='-lpthread -lm' --bindir=/ffmpeg-static/build/20240813/bin --disable-autodetect --disable-everything --disable-doc --enable-gpl --enable-libx264 --enable-decoder='h264,aac,mp2,mp3,pcm_u8'  ...
```
I guess this was here before so not necessarily a question about this change but more generally: How come we have ffmpeg stuff in this python-gis lib?

dotysan replied:
Good point. They were combined for simplicity in the one case where I needed both.

I've removed the ffmpeg stuff from this python-gis image. And will update https://github.com/dotysan/ffmpeg-static/tree/fireview and refer to it in a new container that needs both.

mpalmerlee reviewed (approved):
Don't understand it much but just had a somewhat unrelated question about ffmpeg

dotysan committed:
Create devcontainer user

 - also install sudo & mc

dotysan committed:
Bump NumPy 2.0 => 2.1

dotysan committed:
Put git/lfs in image since we're now a devcontainer

mpalmerlee commented on Dockerfile:
```diff
@@ -270,4 +396,10 @@ ENV PYTHONUNBUFFERED=yessir
+RUN useradd --create-home --shell=/bin/bash gisuser
+# should be UID:GID == 1000:1000
+RUN echo 'gisuser ALL=(ALL) NOPASSWD:ALL' >/etc/sudoers.d/gisuser
+USER gisuser
```
I'm trying to use this base image in R2's AiHeatCleaner and I'm hitting an issue where `gunicorn` isn't found even though it is in the requirements.txt, I suspect because of this change. If I shell into the container I see:
`WARNING: The script gunicorn is installed in '/home/gisuser/.local/bin' which is not on PATH.`

Is the best way to handle this adding this line to my dockerfile you think?
`RUN export PATH="/home/gisuser/.local/bin:$PATH"`

mpalmerlee replied:
That actually didn't seem to work, the best way I've found is like so:
```
# switch to root to install dependencies
USER root

RUN pip install -r requirements.txt

# switch back to gisuser
USER gisuser

CMD gunicorn --bind :$PORT --workers 1 --threads 8 --timeout 0 main:app
```

dotysan replied:
I saw this too, and have a fix someone on an uncommited branch. hope to have it committed once I complete testing of gdal 3.9.2 with PDFium...

IIRC, you can try `sudo pip install ---requirements` so it installs system-wide, not just as gisuser.

mpalmerlee replied:
Oh yeah that makes sense, I'll try that

dotysan replied:
They seem six vs half-dozen. :-)

mpalmerlee replied:
Yeah, not sure why chat GPT didn't suggest that, seems simplest... 🤷

dotysan committed:
indent conditional commands

dotysan committed:
Remove ffmpeg from python-gis

 - some pip updates occurred

dotysan committed:
Temporarily remove the awesome JPEG-XL driver

 - it will be back!

dotysan committed:
Bump GeoPandas to 1.0!

 - also minor folium bump

dotysan committed:
Fix major bug in translating PDF to Tiff with PDFium

 - no errors on compile or translate; but no rasters rendered
 - had to compile both PDFium & GDAL with gcc-11
 - also bump PDFium rev & patch GDAL from git src
 - go back to using system proj for now
 - it was safe to enable JPEG-XL against system lib
 - jettison ffmpeg-static from this image

dotysan committed:
Separate comtainer registries so they can push in parallel

dotysan committed:
Un hard-code github and docker hub users
